plugin-test/src/test/groovy/org/gradle/github/dependency/extractor/BaseExtractorTest.groovy

```diff
@@ -167,6 +150,51 @@ abstract class BaseExtractorTest extends Specification {
         return manifest
     }
 
+    protected List<String> getManifestNames() {
+        return jsonManifests().keySet() as List
+    }
+
+    protected GitHubManifest gitHubManifest(String manifestName) {
+        def jsonManifest = jsonManifest(manifestName)
+        return new GitHubManifest(jsonManifest)
+    }
+
+    protected static class GitHubManifest {
+        Map manifestData
+
+        GitHubManifest(Map manifestData) {
+            this.manifestData = manifestData
+        }
+
+        def getName() {
+            return manifestData.name
+        }
+
+        def getSourceFile() {
+            return (manifestData.file as Map).source_location
+        }
+
+        def getResolved() {
+            def resolved = manifestData.resolved as Map
+            return resolved.keySet() as List
+        }
+
+        def resolved(String key) {
+            def resolved = manifestData.resolved as Map<String, Map>
+            return resolved[key]
+        }
```

Comment:
Do you want to add assertions within each of these to sanity check that each of these values being pulled are actually present before returning them? For example, asserting that the fields being accessed aren't null, and that they have the correct type?

Also, if you know the return type, it would be good to include the return type explicitly.